docs/proposals/scheduling/20220722-multi-hierarchy-elastic-quota-management.md

```diff
@@ -0,0 +1,365 @@
+
+3. We define the "runtime" as the current actual resource that can be used by the quota group. For a quota group whose "request" 
+is less than min, the value of "runtime" is equal to "request". That is to say "request" should be unconditionally satisfied 
+if the "request" is less than "min". For a quota group whose "request" is greater than "min", the value of "runtime" is between 
+"min" and "max", and the part exceeding "min" is based on its own "request", the resources that can be lent, the resources 
+that need to be borrowed, and the ability of other quota groups to compete for borrowing resources. This will be described in 
+detail below.
+
+4. Hierarchy is very important in a resource-shared cluster. Suppose that the cluster shared by multiple departments, and 
+each department has multiple teams. If each team is a quota group, we naturally hope that the relationship between departments 
+and teams is tree shaped. In this way, no matter how to add, delete or adjust quota groups within the department, it is an 
+internal matter of the department. The cluster administrator only needs to be responsible for the quota configuration at the 
+level of departments, and the quota group's configuration can delegate power to the department itself. Moreover, tree can 
+help us easily see the summary of resources from the perspective of departments when there are lots of teams in one department.
+
+Another advantage of tree structure is that we can control the scope of resource borrowing. For example, a department only 
+wants to its quota groups can borrow resources from each other, while the resources of the department do not want to be lent 
+to other departments. This is very convenient for the tree structure. It should be pointed out that although two levels can 
+meet most scenarios (the more levels, the higher the maintenance complexity), we will support that the height of the quota-tree 
+is arbitrary.
+
+### Implementation Details
+
+#### Calculate RuntimeQuota
+
+We use an example to introduce how to calculate "runtime". Suppose the cluster total resource is 100, and has 4 quotas,
+the configuration and "request" of each quota group described as below:
+
+![image](/docs/images/runtimequota1.jpg)
+
+We first calculate the "min" part of "runtime". It should be like as below:
+
+![image](/docs/images/runtimequota2.jpg)
+
+Then we find quota groupA can lent 5 quotas to B\C\D, and the cluster has 40 quotas to allocate, so the sum is 45 for B\C\D 
+to share. We take the proportion of "max" as the basis for B\C\D to allocate shared resources by default, because the larger 
+the "max" is, the more resources it want. In this way, "max" has both the meaning of resource ceiling and allocation proportion. 
+At the same time, we also support using new fields to represent the allocation proportion, which is called "shareRatio". 
+
+In the above example, we assume that "shareRatio" is equal to "max". According to "max" proportion, quota groupB can get
```

Comment:
What scenarios need to set share-ratio separately? Please add relevant descriptions.

Reply:
done